influxql/functions.go

```diff
@@ -53,3 +53,75 @@ func (r *IntegerMeanReducer) Emit() []FloatPoint {
 		Aggregated: r.count,
 	}}
 }
+
+type FloatMovingAverageReducer struct {
+	pos    int
+	sum    float64
+	buf    []float64
+	points []FloatPoint
+}
+
+func NewFloatMovingAverageReducer(n int) *FloatMovingAverageReducer {
+	return &FloatMovingAverageReducer{
+		buf: make([]float64, 0, n),
+	}
+}
+
+func (r *FloatMovingAverageReducer) AggregateFloat(p *FloatPoint) {
+	if len(r.buf) != cap(r.buf) {
+		r.buf = append(r.buf, p.Value)
+	} else {
+		r.sum -= r.buf[r.pos]
+		r.buf[r.pos] = p.Value
+	}
+	r.sum += p.Value
+	r.pos = (r.pos + 1) % cap(r.buf)
+
+	if len(r.buf) == cap(r.buf) {
+		r.points = append(r.points, FloatPoint{
+			Value:      r.sum / float64(len(r.buf)),
+			Time:       p.Time,
+			Aggregated: uint32(len(r.buf)),
+		})
+	}
+}
+
+func (r *FloatMovingAverageReducer) Emit() []FloatPoint {
+	return r.points
```

Comment:
Why are we returning a set of points? Wouldn't we just return a single point with the current average?

Ah I think I get it, the Aggregate method is called exactly `n` times so we only emit one point?